### ISSUE TITLE
why not integrate tfjs-node and tfjs-node-gpu?

### ISSUE THREAD

**Issue by brannondorsey** [tensorflow/tfjs#875]
#### TensorFlow.js version
tfjs-node[-gpu] v0.1.19

#### Browser version
N/A. Node.js or Electron environments.

#### Describe the problem or feature request
Recent activity suggests that tfjs-node-gpu will now fallback to tfjs-node behavior if no CUDA environment + GPU is found. In versions v0.1.17 and v0.1.18 I experienced issues with this fallback (see https://github.com/tensorflow/tfjs/issues/622#issuecomment-428970089 and https://github.com/tensorflow/tfjs/issues/779), but it seems these issues may have stabilized in v0.1.19. **Is this true?**, If so, and tfjs-node-gpu does now fallback to equivalent behavior of tfjs-node, is there any reason to keep the two as separate projects, at least from a user's perspective?

The only thing I can think of is download size: It may not make since to have a user download the GPU-specific libraries if they don't have a GPU. That, and perhaps also some sort of control over whether the GPU is utilized at all (in case the user has a GPU installed but would prefer not to use it). But other than that, is there any reason users should `require("tfjs-node")` over `require("tfjs-node-gpu")`? As an application developer, it seems useful to include one line of code that will work independent of the user's hardware (e.g. `require(`tfjs-node-gpu`)`), instead of adding conditional checks for which version to require.

P.S. Is there a changelog for tfjs-node* that I'm missing somewhere? I've found a bit challenging to grok changes through reading github issues and commits. Not a criticism, just think that would be helpful for those of us that aren't touching the code regularly :smile_cat:

**Comment by brannondorsey** [comment]
Giving this a second look, it seems that https://github.com/tensorflow/tfjs/issues/622#issuecomment-428970089 and https://github.com/tensorflow/tfjs/issues/622#issuecomment-428978656 may not yet be fixed in v0.1.19 after all. 

```javascript
// running on a machine without CUDA + GPU
require("node-tfjs-gpu")
// spits a warning, and then

// reports "cpu" or "webgl" depending on if it's run in Node.js or Electron
tf.getBackend()
```

According to https://github.com/tensorflow/tfjs/issues/622, this fallback should already be working correctly, but I have *never* experienced this to be true. Perhaps it's "working" but the `tf.getBackend()` names are wrong?

**Comment by nkreeger** [comment]
We're not considering this at this time - the GPU support of TensorFlow contains both kernels for CUDA and CPU ops and is very large compared to the CPU-only build of TensorFlow. For this reason we won't be combining them.
